src/util.js

```diff
@@ -0,0 +1,32 @@
+const fs = require('fs')
+const YAML = require('js-yaml')
+const {
+  isFunction, includes, set, isObject,
+} = require('lodash')
+
+const isObj = value => isObject(value) && !isFunction(value)
+
+const KEYS_TO_IGNORE = ['before', 'after']
+
+
+function evaluateData(data) {
+  const keys = Object.keys(data)
+  return keys.reduce((obj, key) => {
+    const value = data[key]
+    const newValue = isFunction(value) && !includes(KEYS_TO_IGNORE, key) ? value() : value
+    const setValue = isObj(newValue) ? evaluateData(newValue) : newValue;
+    set(obj, key, setValue)
+    return obj
+  }, data)
+}
+
+function loadYamlFile(filePath) {
+  const yamlData = evaluateData(YAML.load(fs.readFileSync(filePath, 'utf8')))
+  return yamlData
```

Comment:
Might want to separate the file load from the yaml load from the evaluate. That way you can add the filename to any exception that fires and re-throw it. Otherwise, if there is a yaml parse error, the dev running the test won't know which file had the problem. Same if evaluate fails.